15_mini_project_work/2_num_guessing_game.py

```diff
@@ -42,11 +46,28 @@ def is_valid(input_string, guess_left_border, guess_right_border):
     return result
 
 
+#  returns integer input, asks untill get valid input
+def get_num_input(input_string):
+    is_string_num = False
+    while not is_string_num:
+        if input_string[0] == MINUS_CHAR and input_string[1:].isdigit() or input_string.isdigit():
+            is_string_num = True
+        else:
+            print(TYPE_ERROR_MESSAGE)
+            input_string = input(BORDER_PROMT)
+
+    is_string_num = int(input_string)
+    return is_string_num
+
+
 def guessing_game_run():
     is_guess_wrong = True
     guess_count = 0
-    guess_left_border = int(input(LEFT_BORDER_PROMPT_MESSAGE))
-    guess_right_border = int(input(RIGHT_BORDER_PROMPT))
+    guess_left_border = input(LEFT_BORDER_PROMPT_MESSAGE)
+    guess_left_border = get_num_input(guess_left_border)
+
+    guess_right_border = input(RIGHT_BORDER_PROMPT)
```

Comment:
Move the initial input to the get_nym_input()
it is logical: you are asking to get input until done

Reply:
Initial input - you mean guess not border right?

Reply:
ok